### DISCUSSION TITLE
[Bug] 419 PAGE EXPIRED when I try to login through 127.0.0.1:8000/admin/login

### DISCUSSION THREAD

**Discussion by zanskorja** [Laravel-Backpack/community-forum#1008] [Q&A (Help)] [answered]
# Bug report

### What I did

I have started Laravel project, created a few migrations, models and seeder, then I installed latest Backpack, I generated CRUDS for all models using the command.

### What I expected to happen

I was expected to have a functional install by default.

### What happened

When opening the admin panel the styling doesn't load properly and I get 419 errors. In the network section I see that the next files get error code 404:

- common.js?56ab10a4ca75
- tabler.js?56ab10a4ca75
- colors.css?56ab10a4ca75
- color-adjustments.css?56ab10a4ca75
- common.css?56ab10a4ca75
- style.css?56ab10a4ca75

![image](https://github.com/Laravel-Backpack/CRUD/assets/113106345/b87433a9-126b-4ca1-8791-865a6d1ad100)

### What I've already tried to fix it

I have tried:

- APP_URL to 127.0.0.1 (.env)
- APP_URL to 127.0.0.1:8000 (.env)
- APP_URL to http://127.0.0.1:8000 (.env)
- php artisan key:generate
- I tried reinstalling backpack, updating everything etc.
- BASSET_DEV_MODE=true (.env)
- php artisan config:clear
- php artisan cache:clear
- php artisan view:clear

### Is it a bug in the latest version of Backpack?

After I run ```composer update backpack/crud``` the bug... is it still there?
Yes, nothing changes

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:
### PHP VERSION:
8.2.10

### PHP EXTENSIONS:
Core, bcmath, calendar, ctype, date, filter, hash, iconv, json, SPL, pcre, random, readline, Reflection, session,
 standard, mysqlnd, tokenizer, zlib, libxml, dom, PDO, openssl, SimpleXML, xml, xmlreader, xmlwriter, curl, fileinfo, gd, mbstring, Phar, pdo_mysql, pdo_pgsql, pgsql, zip

### LARAVEL VERSION:
11.9.1.0

### BACKPACK PACKAGE VERSIONS:
backpack/basset: 1.3.4
backpack/crud: 6.7.15
backpack/generators: v4.0.5
backpack/theme-tabler: 1.2.10

**Comment by welcome** [comment]
Hello there! Thanks for opening your first issue on this repo!

Just a heads-up: **Here at Backpack we use Github Issues only for tracking bugs**. Talk about new features is also acceptable. This helps _a lot_ in keeping our focus on improving Backpack. **If you issue is not a bug/feature, please help us out by closing the issue yourself and posting in the appropriate medium (see below)**. If you're not sure where it fits, it's ok, a community member will probably reply to help you with that.

Backpack communication channels:
 - Bug Reports, Feature Requests - Github Issues (here);
 - Quick help (_How do I do X_) - [Gitter Chatroom](https://gitter.im/BackpackForLaravel/Lobby);
 - Long questions (_I have done X and Y and it won't do Z wtf_) - [Stackoverflow](https://stackoverflow.com/questions/tagged/backpack-for-laravel), using the ```backpack-for-laravel``` tag;
 - Showing off something you've made, asking for opinion on Backpack/Laravel matters - [Reddit](https://www.reddit.com/r/BackpackForLaravel/);

Please keep in mind Backpack offers no official / paid support. Whatever help you receive here, on Gitter, Slack or Stackoverflow is thanks to our awesome _awesome_ community members, who give up some of their time to help their peers. If you want to join our community, just start pitching in. We take pride in being a welcoming bunch.

Thank you!

--
Justin Case
The Backpack Robot

**Comment by jcastroa87** [comment]
Hello @zanurban 

If you are using `php artisan serve` to serve your application in `http://127.0.0.1:8000` that's what you need to use as `APP_URL=http://127.0.0.1:8000` in your `.env` file.

If you are using https also make sure you write it down in app url.

Then clean with
```
php artisan optimize:clear
php artisan basset:clear
```

In case this didnt work, check permissions:
```
chmod -R 755 storage
chmod -R 755 vendor
chmod -R 644 bootstrap/caches
```

And take a look in you error logs.

Cheers.

**Comment by zanskorja** [comment]
Hello @jcastroa87 
so my `env` is seted to `http://127.0.0.1:8000`

I tried using commands 

> `php artisan optimize:clear
php artisan basset:clear`

Can I use chmod on Windows.
I am developing my Laravel project on my computer and then I will publish it to my server.

**Comment by jcastroa87** [comment]
Hi.

If you running Windows i dont think you can use these linux commands.

Did you try
```
php artisan basset:check
```

To check if your config is well setting.

Let me know if there is some detail in logs.

Cheers.

**Comment by zanskorja** [comment]
It says that everything is ok

**Comment by zanskorja** [comment]
If I don't do `php artisan serve`,  but if I know correctly, I should run it
Then I get log:
   INFO  Checking Backpack Basset installation.

  Initializing basset check .............................................................................. DONE  
  Checking cache storage ................................................................................. DONE  
  Fetching a basset ..................................................................................... ERROR  

   ERROR  cURL error 7: Failed to connect to 127.0.0.1 port 8000 after 2024 ms: Connection refused (see https://c
url.haxx.se/libcurl/c/libcurl-errors.html) for http://127.0.0.1:8000/storage/basset/vendor/backpack/basset/tests/Helpers/basset-test.js.
  │ Backpack Basset failed to check it's working properly.
  │
  │ This may be due to multiple issues. Please ensure:
  │  1) APP_URL is correctly set in the .env file.
  │  2) Your server is running and accessible at http://127.0.0.1:8000.
  │  3) The public disk is properly configured in config/filesystems.php.
  │     Optionally, basset provides a disk named "basset", you can use it instead.
  │  4) The storage symlink exists and is valid (by default: public/storage).
  │
  │ For more information and solutions, please visit the Backpack Basset FAQ at:
  │ https://github.com/laravel-backpack/basset#faq

**Comment by jcastroa87** [comment]
How did you serve your aplication? Are you using some web server?

**Comment by zanskorja** [comment]
No I am serving it on my machine via `php artisan serve`
And if then I run `php artisan basset:check`
Everything is ok

**Comment by jcastroa87** [comment]
Ok, and wich URl file are trying to load? per example common.js?

Cheers.

**Comment by zanskorja** [comment]
Yes it is trying to load
common.js...
You can see it on picture.
> ![image](https://github.com/Laravel-Backpack/CRUD/assets/113106345/d503977f-fe5e-4e9f-8a95-923efffdbf4d)
And if I click login button I get 419 Page expired

**Comment by jcastroa87** [comment]
Im asking for the URL you write in browser and the full URL of common.js

**Comment by zanskorja** [comment]
Oh sorry, here it is:
http://127.0.0.1:8000/storage/basset/vendor/backpack/crud/src/resources/assets/js/common.js?56ab10a4ca75

**Comment by jcastroa87** [comment]
Look normal, didn you set early storage link?
```
php artisan storage:link
```

**Comment by zanskorja** [comment]
So I tried pulling my private repository to mine other PC and now I have common.js included without errors, Code status: 200, but now when I click login it still does not work now I get on login error:
![image](https://github.com/Laravel-Backpack/CRUD/assets/113106345/922a4806-209b-47b1-b2a5-c16ac17727fd)


> Look normal, didn you set early storage link?
> 
> ```
> php artisan storage:link
> ```

Yea I seted storage link. I just reinstalled Backpack for the third time.

**Comment by zanskorja** [comment]
Could it be CORS policy by any chance?
That CRSF token is not generated properly?

**Comment by jcastroa87** [comment]
CORS is because didnt load common.js

**Comment by jcastroa87** [comment]
You didnt update any config con filesystems.php? Can you share your .env and filesystems.php please, delete all the filesystems.php

Thanks.

**Comment by zanskorja** [comment]
ENV file:
`
APP_NAME=Laravel
APP_ENV=local
APP_KEY=base64:cresLWJ0p1JHNeUSH78lbfzq1PYNkwplo6bDvsf5+LA=
APP_DEBUG=true
APP_TIMEZONE=UTC
APP_URL=http://127.0.0.1:8000

APP_LOCALE=si
APP_FALLBACK_LOCALE=en
APP_FAKER_LOCALE=hr_HR

APP_MAINTENANCE_DRIVER=file
APP_MAINTENANCE_STORE=database

BCRYPT_ROUNDS=12

LOG_CHANNEL=stack
LOG_STACK=single
LOG_DEPRECATIONS_CHANNEL=null
LOG_LEVEL=debug


DB_CONNECTION=pgsql
DB_HOST=localhost
DB_PORT=5432
DB_DATABASE=postgres
DB_USERNAME=postgres
DB_PASSWORD=root

SESSION_DRIVER=database
SESSION_LIFETIME=120
SESSION_ENCRYPT=false
SESSION_PATH=/
SESSION_DOMAIN=null

BROADCAST_CONNECTION=log
FILESYSTEM_DISK=local
QUEUE_CONNECTION=database

CACHE_STORE=database
CACHE_PREFIX=

MEMCACHED_HOST=localhost

CACHE_DRIVER=redis

REDIS_CLIENT=phpredis
REDIS_HOST=localhost
REDIS_PASSWORD=null
REDIS_PORT=6379

MAIL_MAILER=log
MAIL_HOST=127.0.0.1
MAIL_PORT=2525
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS="hello@example.com"
MAIL_FROM_NAME="${APP_NAME}"

AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1
AWS_BUCKET=
AWS_USE_PATH_STYLE_ENDPOINT=false

VITE_APP_NAME="${APP_NAME}"

BASSET_DEV_MODE=true
`

**Comment by zanskorja** [comment]
Filestystems.php I only added my own local disk:

`<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Default Filesystem Disk
    |--------------------------------------------------------------------------
    |
    | Here you may specify the default filesystem disk that should be used
    | by the framework. The "local" disk, as well as a variety of cloud
    | based disks are available to your application for file storage.
    |
    */

    'default' => env('FILESYSTEM_DISK', 'local'),

    /*
    |--------------------------------------------------------------------------
    | Filesystem Disks
    |--------------------------------------------------------------------------
    |
    | Below you may configure as many filesystem disks as necessary, and you
    | may even configure multiple disks for the same driver. Examples for
    | most supported storage drivers are configured here for reference.
    |
    | Supported Drivers: "local", "ftp", "sftp", "s3"
    |
    */

    'disks' => [

        'local' => [
            'driver' => 'local',
            'root' => storage_path('app'),
            'throw' => false,
        ],

        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
            'throw' => false,
        ],

        'localFiles' => [
            'driver' => 'local',
            'root' => storage_path('app/localFiles'),
            'throw' => false,
        ],
    ],

    /*
    |--------------------------------------------------------------------------
    | Symbolic Links
    |--------------------------------------------------------------------------
    |
    | Here you may configure the symbolic links that will be created when the
    | `storage:link` Artisan command is executed. The array keys should be
    | the locations of the links and the values should be their targets.
    |
    */

    'links' => [
        public_path('storage') => storage_path('app/public'),
    ],

];`

**Comment by pxpm** [comment]
Hey @zanurban thanks for the report. 

I am here trying to reproduce your issue but I wasn't able until now. 

![image](https://github.com/Laravel-Backpack/CRUD/assets/7188159/d403a1c6-6d45-4146-9af9-cc322e9ad3e6)

I am more convinced that this is an issue with your setup, because in one of your previous messages you said you were able to load the common.js when moved the project to other PC. 

So I would try to figure out the issue there, and fix it in the other environment. 

When you say you get 419 (expired/not present token), it's when you access the login page, or when you enter the credentials and try to login ? 

Let's also do the following test. If you set `BASSET_DEV_MODE=false` in your env. and issue a `basset:cache` command, does those files get created in `storage/app/public/basset` ? 

You may have your project installed on a directory that's not script writable? I usually put my projects in a folder like `c:/projects/`. I had problems when using projects from example in my USER home folder. 

Let me know if it works. 

Cheers

**Comment by zanskorja** [comment]
> When you say you get 419 (expired/not present token), it's when you access the login page, or when you enter the credentials and try to login ?

I get 419 when I try to log in.

> Let's also do the following test. If you set BASSET_DEV_MODE=false in your env. and issue a basset:cache command, does those files get created in storage/app/public/basset ?

When I go to 127.0.0.1/admin I get Error 500
`cURL error 60: SSL certificate problem: unable to get local issuer certificate (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://unpkg.com/@tabler/core@1.0.0-beta19/dist/css/tabler.min.css`
I think that files does get created:
![image](https://github.com/Laravel-Backpack/CRUD/assets/113106345/52d46597-8951-4385-8603-2b5ead512ccb)

**Reply by fronbow** [reply]
That cURL error is common on Windows boxes. Try the solution here:
[https://sendlayer.com/docs/curl-error-60-ssl-certificate-problem/#:~:text=To%20resolve%20cURL%20error%2060,a%20directory%20on%20your%20server.](https://sendlayer.com/docs/curl-error-60-ssl-certificate-problem/#:~:text=To%20resolve%20cURL%20error%2060,a%20directory%20on%20your%20server.)

That should solve that problem for you :)

**Reply by zanskorja** [reply]
Yes that solved the problem with error code 60 of cURL, but I still get when I hit login button on http://127.0.0.1:8000/admin/login error code 419
![image](https://github.com/Laravel-Backpack/community-forum/assets/113106345/3ecc6b05-bf9f-430b-9472-f70b8dc7b60d)

**Reply by jcastroa87** [reply]
If you are using XAMPP, why you access with port 8000? i still not understand your server config. Can you give us some details about how really did you run you server.

Thanks.

**Comment by tabacitu** [comment]
Maybe SSL isn't properly set up in your Laravel project? Are you getting a green lock icon in the browser? Shouldn't you set the APP_URL to also use HTTPS?

**Reply by tabacitu** [reply]
In any case, I've turned this into a discussion, since I'm pretty sure this is not a bug, but a niche server setup problem or something.

**Comment by fronbow** [comment]
Hi @zanurban 
Have you tried a local server setup - [https://laragon.org/](https://laragon.org/) is my setup of choice on windows and is dead easy to setup - you'll be able to run other tools on the local webserver too for debugging purposes. For instance having [telescope](https://github.com/laravel/telescope) running in another tab so you can check the requests your application is sending. HTH

**Comment by jcastroa87** [comment]
Hello @zanurban did you have any chance to check you server setup to handle about this issue?

Cheers.

**Reply by zanskorja** [reply]
I cannot fix my issue still same. I said as @fronbow said and I now don't get cURL issue any more, but now i get 419 error as previous

**Comment by fronbow** [comment, marked as answer]
Hi @zanurban 

From what I understand, you're either running your web server via the artisan command or through JetBrains Web or PHPStorm?
In which case, they are running PHP's inbuilt web server ( see [https://www.php.net/manual/en/features.commandline.webserver.php](https://www.php.net/manual/en/features.commandline.webserver.php) )
This looks to be less of a Backpack problem and more of a problem with your PHP setup.
For better debugging I would suggest my previous reply about setting up laragon rather than using php's webserver. From my own experience, it's easier to debug a fully formed web server (apache, nginx, etc) than the php builtin as as far as I'm aware the php one is more for testing and debugging.

If you read the php page you will notice this sentence at the top 

> The web server runs only one single-threaded process, so PHP applications will stall if a request is blocked.

This, to me, would suggest why you are getting a 419 status code. Laravel/Backpack generally seems to need to make more than one request at a time and the 419 status code is a client side error code implying that something (the session) has expired.

I would seriously think about installing a local web server, Laragon, WAMP, etc rather than trying to run everything through php's simple webserver.

HTH

**Reply by zanskorja** [reply]
Thank you, I had some troubles setting up Laragoon, but now it works.

**Comment by jcastroa87** [comment]
Im glad you solve it, thanks @fronbow for your help too.

I will close the issue, but please feel free to re-open or create a new one if needed.

Cheers.
